File: smart_plant/src/main/java/com/smart_plant/smart_plant/entity/DiseaseControl.java
package com.smart_plant.smart_plant.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 病虫害防治措施实体。
 *
 * <p>该实体对应数据库中的 disease_control 表，用于保存某个病虫害的预防或治疗措施。</p>
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class DiseaseControl {

    /** 防治措施主键ID。 */
    private Long id;

    /** 关联的病虫害ID，对应 disease_pest.id。 */
    private Long diseaseId;

    /** 病虫害名称，列表展示时通过关联 disease_pest 表带出。 */
    private String diseaseName;

    /** 防治类型：1预防，2治疗。 */
    private Integer controlType;

    /** 防治手段：1农业防治，2物理防治，3化学防治，4生物防治。 */
    private Integer controlCategory;

    /** 措施内容，描述具体预防或治疗方案。 */
    private String method;

    /** 药剂名称，可为空，适用于需要记录用药的措施。 */
    private String drugName;

    /** 使用方法，描述药剂或措施的操作方式。 */
    private String usageMethod;

    /** 剂型、浓度或稀释倍数，例如5%乳油1500倍液。 */
    private String dosageSpec;

    /** 药剂安全间隔期，单位为天。 */
    private Integer safetyIntervalDays;

    /** 禁限用、轮换用药和施药安全等注意事项。 */
    private String precautions;

    /** 适用阶段，例如苗期、花期、结果期等。 */
    private String suitableStage;

    /** 状态：1启用，0停用。 */
    private Integer status;

    /** 展示顺序，数值越小越靠前。 */
    private Integer sortOrder;

    /** 备注信息，用于记录补充说明。 */
    private String remark;

    /** 创建时间，由数据库默认维护。 */
    private LocalDateTime createTime;

    /** 更新时间，由数据库在更新时自动维护。 */
    private LocalDateTime updateTime;
}

File: smart_farm/src/api/diseaseControl.ts
import { get, http, post } from '@/utils/request'
import type { PageInfo } from '@/api/user'

/**
 * 防治措施实体，对应 smart_plant 后端 disease_control 表返回字段。
 * diseaseName 是后端关联 disease_pest 表后带出的展示字段。
 */
export interface DiseaseControl {
  id?: number
  diseaseId: number
  diseaseName?: string
  controlType?: number
  controlCategory?: number
  method?: string
  drugName?: string
  usageMethod?: string
  dosageSpec?: string
  safetyIntervalDays?: number
  precautions?: string
  suitableStage?: string
  status?: number
  sortOrder?: number
  remark?: string
  createTime?: string
  updateTime?: string
}

/**
 * 新增防治措施时提交给后端的字段。
 * createTime、updateTime、diseaseName 等展示字段不参与新增。
 */
export interface DiseaseControlPayload {
  diseaseId: number
  controlType?: number
  controlCategory?: number
  method?: string
  drugName?: string
  usageMethod?: string
  dosageSpec?: string
  safetyIntervalDays?: number
  precautions?: string
  suitableStage?: string
  status?: number
  sortOrder?: number
  remark?: string
}

export interface UpdateDiseaseControlPayload extends Partial<DiseaseControlPayload> {
  id: number
}

export interface DiseaseControlListParams {
  diseaseId?: number
  diseaseName?: string
  controlType?: number
  controlCategory?: number
  status?: number
  pageNum: number
  pageSize: number
}

interface ApiResult<T> {
  code: number
  message: string
  data: T
}

/** 分页查询防治措施列表，支持按病虫害名称、防治类型、防治手段和状态筛选。 */
export const listDiseaseControls = (params: DiseaseControlListParams) => {
  return get<ApiResult<PageInfo<DiseaseControl>>>(
    '/disease-control/list',
    params as unknown as Record<string, unknown>,
  )
}

/** 根据 ID 查询防治措施详情，供编辑弹框回填。 */
export const getDiseaseControlById = (id: number) => {
  return get<ApiResult<DiseaseControl>>(`/disease-control/${id}`)
}

/** 新增防治措施。 */
export const addDiseaseControl = (data: DiseaseControlPayload) => {
  return post<ApiResult<DiseaseControl>>(
    '/disease-control/add',
    data as unknown as Record<string, unknown>,
  )
}

/** 编辑防治措施。 */
export const updateDiseaseControl = (data: UpdateDiseaseControlPayload) => {
  return http<ApiResult<DiseaseControl>>({ url: '/disease-control', method: 'put', data })
}

/** 删除单条防治措施。 */
export const deleteDiseaseControl = (id: number) => {
  return http<ApiResult<void>>({ url: `/disease-control/${id}`, method: 'delete' })
}

/** 批量删除防治措施。 */
export const batchDeleteDiseaseControls = (ids: number[]) => {
  return http<ApiResult<number>>({ url: '/disease-control/batch', method: 'delete', data: ids })
}

/** 启用或停用防治措施。 */
export const updateDiseaseControlStatus = (id: number, status: number) => {
  return http<ApiResult<void>>({
    url: `/disease-control/${id}/status`,
    method: 'put',
    params: { status },
  })
}
